Tests/GRPCTests/ServerErrorDelegateTests.swift

```diff
@@ -0,0 +1,147 @@
+
+  func testTransformLibraryError_whenTransformingErrorToStatus_bidirectionalStreaming() throws {
+    try testTransformLibraryError_whenTransformingErrorToStatus(uri: "/echo.Echo/Update")
+  }
+
+  private func testTransformLibraryError_whenTransformingErrorToStatus(uri: String) throws {
+    self.setupChannelAndDelegate { _ in
+      GRPCStatusAndMetadata(status: .init(code: .notFound, message: "some error"))
+    }
+    let requestHead = HTTPRequestHead(
+      version: .init(major: 2, minor: 0),
+      method: .POST,
+      uri: uri,
+      headers: ["content-type": "application/grpc"]
+    )
+
+    XCTAssertNoThrow(try self.channel.writeInbound(HTTPServerRequestPart.head(requestHead)))
+    self.channel.pipeline.fireErrorCaught(GRPCStatus(code: .aborted, message: nil))
+    // This is the head
+    XCTAssertNoThrow(try self.channel.readOutbound(as: HTTPServerResponsePart.self))
+    let end = try self.channel.readOutbound(as: HTTPServerResponsePart.self)
+
+    guard case let .end(headers) = end else {
+      XCTFail("Expected headers but got \(end.debugDescription)")
+      return
+    }
+
+    XCTAssertEqual(headers?.first(name: "grpc-status"), "5")
+    XCTAssertEqual(headers?.first(name: "grpc-message"), "some error")
+  }
+
+  func testTransformLibraryError_whenTransformingErrorToStatusAndMetadata_unary() throws {
+    try testTransformLibraryError_whenTransformingErrorToStatusAndMetadata(uri: "/echo.Echo/Get")
+  }
+
+  func testTransformLibraryError_whenTransformingErrorToStatusAndMetadata_clientStreaming() throws {
+    try testTransformLibraryError_whenTransformingErrorToStatusAndMetadata(uri: "/echo.Echo/Collect")
+  }
+
+  func testTransformLibraryError_whenTransformingErrorToStatusAndMetadata_serverStreaming() throws {
+    try testTransformLibraryError_whenTransformingErrorToStatusAndMetadata(uri: "/echo.Echo/Expand")
+  }
+
+  func testTransformLibraryError_whenTransformingErrorToStatusAndMetadata_bidirectionalStreaming() throws {
+    try testTransformLibraryError_whenTransformingErrorToStatusAndMetadata(uri: "/echo.Echo/Update")
+  }
+
+  private func testTransformLibraryError_whenTransformingErrorToStatusAndMetadata(uri: String) throws {
+    self.setupChannelAndDelegate { _ in
+      GRPCStatusAndMetadata(
+        status: .init(code: .notFound, message: "some error"),
+        metadata: ["some-metadata": "test"]
+      )
+    }
+    let requestHead = HTTPRequestHead(
+      version: .init(major: 2, minor: 0),
+      method: .POST,
+      uri: uri,
+      headers: ["content-type": "application/grpc"]
+    )
+
+    XCTAssertNoThrow(try self.channel.writeInbound(HTTPServerRequestPart.head(requestHead)))
+    self.channel.pipeline.fireErrorCaught(GRPCStatus(code: .aborted, message: nil))
+    // This is the head
+    XCTAssertNoThrow(try self.channel.readOutbound(as: HTTPServerResponsePart.self))
+    let end = try self.channel.readOutbound(as: HTTPServerResponsePart.self)
+
+    guard case let .end(headers) = end else {
```

Comment:
Same here; I _think_ we need `.some(.end(headers))` to make this Swift 5.0 friendly